### PR TITLE
hbiperftest: added a new puptoo template, added few helper functions that is useful for the hbiperftests, tally.

### PR DESCRIPTION
**Pull request by vinaykns**
that is useful for the hbiperftests, tally.

### CODE REVIEW

File: opl/db.py [hunk 1]
```diff
@@ -7,6 +7,40 @@
 import psycopg2.extras
 
 
+# get_query_result is a wrapper function that gives query result with providing database
```
Comment (jhutar) [inline opl/db.py:10]: Please move docs string into the function:
```
# this does that
def this_does():
```
=>
```
def this_does():
    """This does that."""
```

File: opl/db.py [hunk 2]
```diff
@@ -7,6 +7,40 @@
...
+# credentials.
+def get_query_result(db_conf, sql):
```
Comment (jhutar) [inline opl/db.py:12]: Hello! Why do we need this function? Usually you do not pass DB hostname, creds..., you just pass DB client object. Feel free to keep it in the PR, just wanted to warn you that this way the script needs to connect to the DB for every query, which takes lots of time when compared to connecting once and just passing client object.

File: opl/db.py [hunk 3]
```diff
@@ -78,6 +112,27 @@ def get_timedelta_between_timestamp_n_dbcolumn(start_time, connection, column, t
     return timedelta
 
 
+def load_distinct_accounts(inventory_db_conf, d_dict):
```
Comment (jhutar) [inline opl/db.py:115]: This is very HBI specific. Please move to `opl/inventory.py` or so.

File: opl/db.py [hunk 4]
```diff
@@ -78,6 +112,27 @@ def get_timedelta_between_timestamp_n_dbcolumn(start_time, connection, column, t
...
+    query = 'select distinct account from hosts;'
+    data_list = get_query_result(inventory_db_conf, query)
+    for account in data_list:
+        d_dict[account]
+
+    return d_dict
+
+
+# get_unique_key generates an account_id which is not present in the input d_dict
+def get_unique_key(d_dict):
```
Comment (jhutar) [inline opl/db.py:125]: Also, this should not be in generic `opl/db.py`.

File: opl/generators/inventory_ingress.py [hunk 5]
```diff
@@ -44,6 +46,18 @@ def __init__(self, count, fraction=1, relatives=100, addresses=3, packages=500,
     def __iter__(self):
         return self
 
+    def _get_service_data(self):
```
Comment (jhutar) [inline opl/generators/inventory_ingress.py:49]: Why do we need this function? Can we create `enabed_services`, `yum_repos`... directly in `_get()` function? E.g.:

```
        data = {
                  'subscription_manager_id': self._get_uuid(),
                  'bios_uuid': self._get_bios_uuid(),
                  'request_id': self._get_uuid(),
                  'owner_id': self._get_uuid(),
                  'insights_id': self._get_uuid(),
                  'ipv4_addr': self._get_ipv4(),
                  'ipv6_addr': self._get_ipv6(),
                  'mac_addr': self._get_mac(),
                  'fqdn': self._get_hostname(),
                  'nowz': self._get_now_iso_z(),   # well, this is in nano-seconds, but should be in mili-seconds
                  'tommorowz': self._get_tommorow_iso_z(),
                  'packages': self.pg.generate(self.packages),
                  'yum_repos': opl.generators.packages.YumReposGenerator().generate(137),
                  'enabled_services': opl.generators.packages.EnabledServicesGenerator().generate(139),
                  'installed_services': opl.generators.packages.InstalledServicesGenerator() .generate(160),
                  'running_processes': opl.generators.packages.RunningProcessesGenerator().generate(89),
        }
```
Or you can create helper functions like:
```
def _get_yum_repos(self):
    return opl.generators.packages.YumReposGenerator().generate(137)
```
and then just have:
```
        data = {
                  'subscription_manager_id': self._get_uuid(),
                  'bios_uuid': self._get_bios_uuid(),
                  'request_id': self._get_uuid(),
                  'owner_id': self._get_uuid(),
                  'insights_id': self._get_uuid(),
                  'ipv4_addr': self._get_ipv4(),
                  'ipv6_addr': self._get_ipv6(),
                  'mac_addr': self._get_mac(),
                  'fqdn': self._get_hostname(),
                  'nowz': self._get_now_iso_z(),   # well, this is in nano-seconds, but should be in mili-seconds
                  'tommorowz': self._get_tommorow_iso_z(),
                  'packages': self.pg.generate(self.packages),
                  'yum_repos': self._get_yum_repos(),
                  ...
        }
```

File: opl/generators/packages.py [hunk 6]
```diff
@@ -16,3 +16,51 @@ def count(self):
     def generate(self, count):
         return [random.choice(self.data[key])
                 for key in random.sample(self.data.keys(), count)]
+
+
+class YumReposGenerator:
```
Comment (jhutar) [inline opl/generators/packages.py:21]: Nice!

**Review by jhutar — CHANGES_REQUESTED** [review]
This looks very good! Please fix few tiny bits and we are ready to go!

**Comment by vinaykns** [comment]
@jhutar  please take another look.

File: opl/generators/inventory_ingress.py [hunk 7]
```diff
@@ -129,8 +151,13 @@ def _get(self):
             'nowz': self._get_now_iso_z(),   # well, this is in nano-seconds, but should be in mili-seconds
             'tommorowz': self._get_tommorow_iso_z(),
             'packages': self.pg.generate(self.packages),
+            'yum_repos': opl.generators.packages.YumReposGenerator().generate(137),
+            'enabled_services': opl.generators.packages.EnabledServicesGenerator().generate(139),
+            'installed_services': opl.generators.packages.InstalledServicesGenerator() .generate(160),
+            'running_processes': opl.generators.packages.RunningProcessesGenerator().generate(89),
         }
         data.update(random.choice(self.relatives))   # add account and orgid
+        data.update({'b64_identity': self._get_b64_identity(data['account'])})
```
Comment (jhutar) [inline opl/generators/inventory_ingress.py:160]: You can put this `'b64_identity': self._get_b64_identity(data['account'])` directly above where we create `data` variable, but this works as well.

**Comment by jhutar** [comment]
Thank you!

**Pull request merged**